components-part1/src/partials/Home/Content.test.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { render, screen, cleanup, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import Content from './Content';
+import httpService from '../../services/httpService';
+import { act } from 'react-dom/test-utils';
+import { moviesList } from '../../mockData/MoviesList';
+
+const mockData = [...moviesList];
+afterEach(cleanup);
+jest.mock('../../services/httpService');
+
+const mockedUsedNavigate = jest.fn();
+const mockUsedLocation = jest.fn();
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useNavigate: () => mockedUsedNavigate,
+  useLocation: () => mockUsedLocation,
+}));
+
+const mockSelector = jest.fn();
+jest.mock('react-redux', () => ({
+  ...jest.requireActual('react-redux'),
+  useSelector: () => mockSelector,
+  useDispatch: () => jest.fn(),
+}));
+
+const handleGenreClick = jest.fn();
```

Comment:
it's better to have mock functions created in setup functions, and clean them in clean-up functions, afterAll and afterEach, to have a clean context for the next tests.